gengokit/middlewares/templates/endpoints.go

```diff
@@ -5,9 +5,20 @@ package middlewares
 
 import (
 	"github.com/go-kit/kit/endpoint"
+	svc "{{.ImportPath -}} /generated"
 )
 
-func InjectEndpointMiddlewares(in endpoint.Endpoint) endpoint.Endpoint {
+// WrapEndpoint will be called individually for all endpoints defined in
+// the service. Implement this with the middlewares you want applied to
+// every endpoint.
+func WrapEndpoint(in endpoint.Endpoint) endpoint.Endpoint {
```

Comment:
None of us has had any great revelations about how this should be named or restructured. I want everyone to be happy with things, but in the name of lot leaving this functional change hanging out to dry, and to keep things moving, I'm going to merge it, and we can deal with it if there is further gnashing of teeth over the way it's been approached. There's certainly a cleaner way that probably only exposes a single function from `endpoints.go`, but it will require thought and discussion that I haven't the luxury of time for right now.

At the end of the day, the user ought not to be a derp, and be capable of reading the GoDoc attached to each function. They will either find the function names intuitive or stupid, and if someone is so moved, they can submit a PR.